File: telegram-bot/src/main/java/idgtl/Catch.java
package idgtl;

import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Pointcut;

/**
 * Created by d.asadullin on 12.10.2016.
 */
@Aspect
public class Catch {

    @Around("execution(* org.telegram.telegrambots.api.objects.MessageEntity.computeText(..))")
    public Object profile(ProceedingJoinPoint pjp) throws Throwable{
        try {
            Object output = pjp.proceed();
            return output;
        }catch (Exception ex) {
            return "";
        }
    }

    public static Catch aspectOf(){
       return new Catch();
    }

}

File: telegram-bot/src/main/java/idgtl/Catch.kt
package idgtl

import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect
import org.aspectj.lang.annotation.Pointcut

/**
 * Created by d.asadullin on 12.10.2016.
 */
@Aspect
class Catch {

    @Around("execution(* org.telegram.telegrambots.api.objects.MessageEntity.computeText(..))")
    @Throws(Throwable::class)
    fun profile(pjp: ProceedingJoinPoint): Any {
        try {
            val output = pjp.proceed()
            return output
        } catch (ex: Exception) {
            return ""
        }

    }
    companion object {

        @JvmStatic fun aspectOf(): Catch {
            return Catch()
        }
    }

}
